Chore: fix failing tests

madcampos opened the pull request:
## Summary

<!-- Describe what problem you are tying to solve -->

## Tests

<!-- Tell us if the changes have been tested and how -->

## Additional information

<!-- Do we need additional context or information about the changes? Please describe it here -->

Resolves: #200, #201, #202, #203, #204, #205, #206

madcampos committed:
fix: #204 team with same name

madcampos committed:
fix: #206 organizers can create teams

madcampos committed:
fix: #204 disalow dupe team name on update

madcampos committed:
fix: #202 only add team members that exist

madcampos committed:
fix: #201 deleting team members with bad ids

madcampos committed:
Merge branch 'chore/update-tests' into chore/fix-failing-tests

madcampos committed:
fix: #201 resolve issue with deleting non member ids

madcampos committed:
fix: #200 resolve data returned by getting all team members

madcampos committed:
fix: invalid team id

madcampos committed:
fix: #201 check ids when updating and deleting team members

madcampos committed:
fix: #203 resolve validation for both name and description when updating team members

madcampos committed:
fix: uncomment fixed tests

madcampos committed:
feat: add checks for data size and for total body size

madcampos committed:
Merge branch 'chore/update-tests' into chore/fix-failing-tests

madcampos committed:
fix: #96 test for strong passwords

madcampos committed:
fix: duplicated team names

madcampos committed:
fix: wrong http method and status code

madcampos committed:
fix: reference the correct field

madcampos committed:
fix: status code

madcampos committed:
fix: remove string destructuring and fix missing query parameter

madcampos committed:
fix: update profile reference

madcampos committed:
chore: ensure project builds before pushing

madcampos committed:
fix: remove extra test failing due to race condition

manvinderjit commented on api/src/middleware/body-size.ts:
```diff
@@ -0,0 +1,26 @@
+/** 48kb */
+const TOTAL_FILE_SIZE_IN_KB = 48 * 1024;
+
+/** 512kb */
+const FILE_UPLOAD_SIZE_IN_KB = 512 * 1024;
```
question: Is this for the whole project? Because in the file upload issue specs, we have specified 5MB limit for upload file.

<img width="351" height="155" alt="image" src="https://github.com/user-attachments/assets/489cf18f-c43e-40b2-9f2c-64a2b8b3fa83" />

madcampos replied:
I'll fix that, the size is wrong here.

manvinderjit commented on api/src/routes/auth/validation.ts:
```diff
@@ -46,11 +50,13 @@ export type Access = z.infer<typeof AccessSchema>;
 	email: z
 		.email('Invalid Email')
+		.max(SHORT_TEXT_SIZE_IN_CHAR)
 		.trim()
```
suggestion: trim before determining min/max

manvinderjit commented on api/src/routes/auth/validation.ts:
```diff
@@ -46,11 +50,13 @@ export type Access = z.infer<typeof AccessSchema>;
 	password: z
 		.string()
+		.max(SHORT_TEXT_SIZE_IN_CHAR)
 		.trim()
```
suggestion: trim before determining min/max

manvinderjit commented on api/src/routes/auth/validation.ts:
```diff
@@ -60,15 +66,18 @@ export type SignInData = z.infer<typeof SignInSchema>;
 	name: z
 		.string({ error: 'Name is required' })
+		.max(SHORT_TEXT_SIZE_IN_CHAR)
 		.trim()
```
suggestion: trim before determining min/max

manvinderjit commented on api/src/routes/event-log/validation.ts:
```diff
@@ -22,6 +23,7 @@ export const EventLogSchema = IdAndSchemaVersionSchema
 			verb: z
 				.string()
+				.max(SHORT_TEXT_SIZE_IN_CHAR)
 				.trim()
```
suggestion: trim before determining min/max

manvinderjit commented on api/src/routes/profile/validation.ts:
```diff
@@ -12,16 +13,19 @@ export const ProfileSchema = BaseDbEntitySchema.extend(
 		email: z
 			.email('Invalid Email.')
+			.max(SHORT_TEXT_SIZE_IN_CHAR)
 			.trim()
```
suggestion: trim before max/min

manvinderjit commented on api/src/routes/team/index.ts:
```diff
@@ -223,6 +239,14 @@ teamRoutes.openapi(
 		}
 
+		if (body.name) {
+			const hasExistingTeamName = await doestSameTeamNameExist(context.env.Database, body.name);
```
typo: 
```suggestion
			const hasExistingTeamName = await doesSameTeamNameExist(context.env.Database, body.name);
```

manvinderjit commented on api/src/routes/team/validation.ts:
```diff
@@ -1,15 +1,18 @@
 		name: z
 			.string()
+			.max(SHORT_TEXT_SIZE_IN_CHAR)
 			.trim()
```
suggestions: trim before min/max

manvinderjit reviewed (dismissed):
@madcampos Thank you for all the effort. I left minor suggestions and couple of clarifications.

madcampos committed:
fix: file upload size and docs for size limits

madcampos committed:
fix: typo

madcampos committed:
feat: update schemas

- Resorganize schema validation ordering so all schemas trim first and then check for min and max values.
- Infer schemas as much as possible so we reuse references instead of create new schemas.
- Add missing descriptions.
- Normalize schema errors to use Zod's default.

madcampos committed:
fix: reference of hearbeat response

madcampos committed:
fix: empty rescription tests now correctly error out

kbventures commented on api/src/middleware/body-size.ts:
```diff
@@ -0,0 +1,26 @@
+import { bodyLimit } from 'hono/body-limit';
+import { StatusCodes, type StatusResponse } from '../utils/responses.ts';
+
+/** 48kb */
```
Nitpick: KiB, not KB.  1KB 1000 bytes, while 1 KiB is 1014bytes.

Question: What is the rationale/source for choosing these specific metrics for protecting the endpoint?

madcampos replied:
Regarding KB vs KiB:

<img width="747" height="1155" alt="Meme from Futurama with the caption: You are technically correct, the best kind of correct." src="https://github.com/user-attachments/assets/73339d17-e742-473c-959c-ea1606b3e1d1" />

While yes, you are correct, I prefer keeping the same naming as in this context, it doesn't make a difference if we are using one unit or another. Plus, people are more used to file sizes being in KB rather than in KiB.

As for the specific values, here are the references:
- https://github.com/torontojs/communityhub/issues/122
- https://github.com/torontojs/communityhub/issues/98#issuecomment-2933030822

kbventures commented on api/src/routes/profile/data.ts:
```diff
@@ -29,6 +29,22 @@ export async function doesProfileExist(database: D1Database, id: string) {
 	return Boolean(profile);
 }
 
+export async function nonExistingProfileIds(database: D1Database, ids: string[]) {
```
Question:

nonExistingProfileIds ???

madcampos replied:
It checks the database for profile ids that do not exist, if you pass a random uuid that is not on the database, this will tell you it is not there. Should the function be renamed?

kbventures commented on api/src/routes/team-members/data.ts:
```diff
@@ -1,7 +1,24 @@
-import type { AddTeamMembers, UpdateTeamMembers } from './validation.ts';
+import type { AddTeamMembers, TeamMemberInfo, UpdateTeamMembers } from './validation.ts';
+
+export async function nonExistingTeamMemberIds(database: D1Database, teamId: string, ids: string[]) {
```
Same as the previous question...nonExistingTeamMembberIds seems contradictory and a question in itself..

kbventures reviewed (approved):
Great work. Some notes around body limit middleware sizes decisions would be nice.
